fastapi/dependencies/utils.py

```diff
@@ -208,15 +208,57 @@ def get_flat_params(dependant: Dependant) -> List[ModelField]:
 def get_typed_signature(call: Callable[..., Any]) -> inspect.Signature:
     signature = inspect.signature(call)
     globalns = getattr(call, "__globals__", {})
-    typed_params = [
-        inspect.Parameter(
-            name=param.name,
-            kind=param.kind,
-            default=param.default,
-            annotation=get_typed_annotation(param.annotation, globalns),
-        )
-        for param in signature.parameters.values()
-    ]
+    if PYDANTIC_V2:
+        fields = getattr(call, "model_fields", {})
+    else:
+        fields = getattr(call, "__fields__", {})
+    if len(fields):
+        alias_dict = {}
+        query_extra_info = {}
+        for param in fields:
+            if PYDANTIC_V2:
+                query_extra_info[param] = dict(fields[param].__repr_args__())
+            else:
+                query_extra_info[param] = dict(fields[param].field_info.__repr_args__())
+
+            if "alias" in query_extra_info[param]:
```

Comment:
There is also `validation_alias` attribute. I think it should also be considered and tested